src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -9,6 +9,25 @@ public class RegistrationServiceImpl implements RegistrationService {
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new RuntimeException("User can not be null");
+        }
+
+        if (user.getLogin() == null || user.getLogin().length() < 6) {
+            throw new RuntimeException("login must contain more than 6 characters");
+        }
+
+        if (user.getPassword() == null || user.getPassword().length() < 6) {
+            throw new RuntimeException("Password must be longer than 6");
+        }
+
+        if (user.getAge() < 18) {
+            throw new RuntimeException("User must be over 18");
+        }
+
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new RuntimeException("Such User already exists");
```

Comment:
The exception message 'Such User already exists' could be more informative. Consider including the user's login in the message to clarify which user is causing the conflict.